packages/muze-legend/src/legend/legend-helper.js

```diff
@@ -27,16 +27,164 @@ export const getScaleInfo = (scale) => {
 };
 
 /**
+  * Function to recompute the stops in array in case of Top and Bottom Alignment of legend.
+  *
+  * @param  {number} availableMaxWidth - max Available Space
+  * @param  {number} requiredWidth - required width
+  * @param  {Array} domainForLegend - array
+  * @param  {Object} smartLabelCalc - smartLabel manager
+  * @return {Array} - Recomputed Array
+  */
+const getHorizontalRecomputedArray = (availableMaxWidth, requiredWidth, domainForLegend, smartLabelCalc) => {
+    // declaring Current and Next Tick Value variable
+    let currentTickValue;
+    let nextTickValue;
+    const upperBound = domainForLegend[domainForLegend.length - 1];
+
+    // calculating pixel required per Tick Values
+    const pixelPerTick = (availableMaxWidth / upperBound);
+
+    // checking if available max width is smaller than required width for legend.
+    if (availableMaxWidth >= requiredWidth) {
+        return domainForLegend;
+    }
+
+    currentTickValue = (smartLabelCalc.getOriSize((domainForLegend[0])).width);
+    for (let i = 1; i < domainForLegend.length; i++) {
+        nextTickValue = ((currentTickValue / pixelPerTick) + (currentTickValue / 2) + domainForLegend[i - 1]);
+        if (domainForLegend[i] < nextTickValue) {
+            domainForLegend.splice(i, 1);
+            i -= 1;
+        }
+        currentTickValue = (smartLabelCalc.getOriSize((domainForLegend[i])).width);
+    }
+    return domainForLegend;
+};
+
+/**
+ * Function to recompute the stops in array in case to left and right alignment of legend.
  *
- *
- * @param {*} domain
- * @param {*} steps
- *
+ * @param  {number} availableMaxHeight - max Available Height
+ * @param  {number} requiredHeight - required height
+ * @param  {Array} domainForLegend - array
+ * @param  {Object} smartLabelCalc - smartLabel Manager
+ * @return {Array} - Recomputed Array
+ */
+const getVerticalRecomputedArray = (availableMaxHeight, requiredHeight, domainForLegend, smartLabelCalc) => {
```

Comment:
Combine horizontal and vertical logic. Only write separate code for separate functions

Reply:
combined the vertical and horizontal alignment logic.